components/build/src/lib.rs

```diff
@@ -152,13 +152,9 @@ fn generate_mod_rs<P: AsRef<Path>, Q: AsRef<Path>>(
 
     // To avoid cases where input sources are also added as includes, use only
     // unique paths.
-    let proto_files: Vec<PathBuf> = [proto_files, includes]
-        .concat()
-        .into_iter()
-        .unique()
-        .collect();
+    let proto_files: HashSet<PathBuf> = [proto_files, includes].concat().into_iter().collect();
```

Comment:
Also btw you can use `proto_files.iter().chain(includes.iter()).collect()` to avoid allocation for concatenated slice.